.prettierrc

```diff
@@ -1,4 +1,4 @@
 {
   "singleQuote": true,
-  "trailingComma": "all"
+  "trailingComma": "es5"
```

Comment:
Well, this was necessary so the right click "formatting" with "prettier-tslint" is the same as the `cli` one. I told you that "prettier" can cause many more troubles than it solves.

Reply:
Notice the `"prettier.tslintIntegration": true,` in `settings.json`.

Reply:
I've been using `prettier` with typescript during the last year and a half and I haven't find any issue yet. That, of course, doesn't mean there may be some problems. What I see here is a new tool, which is `prettier-tslint` which tries to mix the both tools. This is the one introducing the friction. The way I use this, is two separate tools. `Prettier` only takes care about formatting and `tslint` about code rules. It shouldn't be a problem at all.

Reply:
Unfortunately for you `prettier` and `tslint` share some formatting rules that don't play well together.
`tslint` rule `"trailing-comma": [true, { "esSpecCompliant": true }]` is not the same as `prettier` rule `"trailingComma": "all"`. `prettier` rule `"trailingComma": "es5"` plays nicer.

Reply:
HAving a trailing comma on the last parameter strikes me odd.

Reply:
I'll look into that. I prefer to have the trailing comma on last parameters if possible. It's really convenient when modifying object's properties and it doesn't count double in git when you add a property (one line for the comma, and one line for the adding). On top of that, it's easier to comment last parameters/properties.

Reply:
In that case, `tslint` rule has to be changed.

Reply:
Something we need to consider is the case where a user is not installing `prettier` and only relies on the `tslint` formatting rules. My effort is to have this case also produce the same formatting result as when using `prettier`. And the conflict is only affecting the trailing comma and the quoting.

With the above comment revised rule the issue seems to be resolved and a triling comma will always be present for multilines.

Reply:
Prettier is a devDependency, or at least it was, and that's why we had the precommit hook to run it every time any dev would try to commit.

Reply:
I was referring to the `vscode` extension, my bad.

Reply:
With the precommit hook, this kind of user will be notified of what he needs to change, or we can even prettify the changes automatically by running prettier.

Reply:
That's the point. I never gotten notified because I commit via a GUI. Never the less, this issue, with `prettierrc` and `tslint`, is resolved as I already mentioned.